sources/core/Stride.Core.CompilerServices/Common/SymbolExtensions.cs

```diff
@@ -13,10 +13,23 @@ public static bool IsVisibleToSerializer(this ISymbol symbol, bool hasDataMember
         var accessibility = symbol.DeclaredAccessibility;
 
         if (hasDataMemberAttribute)
-            return accessibility == Accessibility.Public || accessibility == Accessibility.Internal || accessibility == Accessibility.ProtectedOrInternal;
+            return accessibility is Accessibility.Public or Accessibility.Internal or Accessibility.ProtectedOrInternal;
         return accessibility == Accessibility.Public;
     }
 
+    /// <summary>
+    /// Tries to get <paramref name="attribute"/> on the <paramref name="symbol"/>
+    /// </summary>
+    /// <param name="symbol">The Symbol to search on</param>
+    /// <param name="attribute">The attribute looking for</param>
+    /// <param name="attributeData">The <see cref="AttributeData"/> of the Attribute if it is found</param>
+    /// <returns>true if the attribute is found, else false</returns>
+    public static bool TryGetAttribute(this ISymbol symbol, INamedTypeSymbol attribute, out AttributeData attributeData)
+    {
+        attributeData = symbol.GetAttributes().FirstOrDefault(attr => attr.AttributeClass?.OriginalDefinition.Equals(attribute, SymbolEqualityComparer.Default) ?? false)!;
+        return attributeData is not null;
+    }
```

Comment:
I think this can be improved, do you have access to [MaybeNullWhen](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.maybenullwhenattribute?view=net-8.0) in this project ? To decorate `AttributeData attributeData` into `[MaybeNullWhen(false)] AttributeData attributeData`
Then:
```cs
if (symbol.GetAttributes().FirstOrDefault(attr => attr.AttributeClass?.OriginalDefinition.Equals(attribute, SymbolEqualityComparer.Default) ?? false) is {} data)
{
	attributeData = data;
	return true;
}

attributeData = null;
return false;
```
You can also change the signature of the function, something like:
```cs
public static AttributeData? TryGetAttribute(this ISymbol symbol, INamedTypeSymbol attribute)
{
    return symbol.GetAttributes().FirstOrDefault(attr => attr.AttributeClass?.OriginalDefinition.Equals(attribute, SymbolEqualityComparer.Default) ?? false);
}
```
And adapt the call site to this new signature instead.

Reply:
https://github.com/stride3d/stride/pull/2104/commits/d8103fc25e74ba1f8aef348b9f693279742e49f0 added